server/seed.py

```diff
@@ -0,0 +1,156 @@
+    SQLAlchemy won't know about the models and therefore will NOT
+    create the proper tables in your database which will cause this
+    function to fail.
+
+    Create data in the database by randomly generating values.
+    Once this function is complete, it will write data that is
+    likely to be used in testing to the 'seeded_objects_info.txt'
+    file with separators and linebreaks to make it easier to read.
+
+    When a user instance is created, its subsequent Profile is created
+    as well and then they are associated with one another.
+    After the user and profile objects are created, create some recipe
+    objects to be associated with the profile object. Leave some profile
+    objects without any recipes for filter testing.
+    """
+
+    # Helper functions
+    def random_value(value_type, length, is_price=False, is_password=False):
+        """
+        Create random values to fill out the data for the random models
+        """
+        if is_price:
+            return random.randint(300, 1000)
+        elif value_type == "int":
+            return random.randint(0,length)
+        elif is_password:
+            # Everyone gets the same password
+            return "AbcdE123"
+        elif value_type == "str":
+            # make a string of the given length and return
+            letters = string.ascii_letters
+            return "".join(random.choice(letters) for letter in range(length))
+    def generate_email():
+        return random_value("str", 10)+"@gmail.com"
+
+    users = []
+    profiles = []
+    recipes = []
+    is_chef = False
+    # Create how ever many user and profile objects the user defined
+    for i in range(num_of_instances):
+
+        # Create user
+        new_user = User.create(**{
+            'street_address': random_value('str', 16),
+            'city': random_value('str', 10),
+            'state_or_province': random_value('str', 15),
+            'country': random_value('str', 10),
+            'zip_code': "45698",
+            'username': random_value('str', 20),
+            'email': generate_email(),
+            # The first two args don't actually matter when is_password=True
+            'password': random_value("str", 15, is_password=True)
+            })
+        users.append(new_user)
+
+        # Create profile     
+        new_profile = Profile()
+        new_profile = new_profile.create(**{
+            'name': random_value('str', 15),
+            'is_chef': is_chef,
+            'about_me': random_value('str', 50),
+            'profile_image': random_value('str', 12),
+            'favourite_cuisine': random.choice(Recipe.get_cuisines()),
+            "location": random_value('str', 20),
+        })
+        profiles.append(new_profile)
+
+        # Assign the new profile to the new user
+        new_user.assign_one_to_one('profile', new_profile)
+        # Make some users normal, i.e no recipes
+        if not is_chef:
+            is_chef = True
+            continue
```

Comment:
It looks like your script is making all of the users chefs, adding recipes and then making them not chefs, is that intentional?

Reply:
Yes, it's intentional. It makes it so every other user is not a chef so there is some diversity in the data so we can test searches and such. Here's an example, if we want to search recipes by availability, you can see the dataset without the filter then you can see how many recipes are not available but if everything were available you'd have to change that by hand to test that feature.
I figured I would try to make the generated data in a way that could make proper test results more obvious so that's why some users are not chefs (and don't have recipes) and some recipes have available=False